tools/coverity/misra.config

```diff
@@ -8,27 +8,20 @@
         {
             deviation: "Directive 4.9",
             category: "Advisory",
-            reason: "Allow inclusion of function like macros."
+            reason: "Allow inclusion of function like macros. The `assert` macro is used throughout the library for parameter validation, and logging is done using function like macros."
```

Comment:
Is this required since assert is disabled ?

Reply:
So there are calls to assert() that set this off, but even with the DNDEBUG there are actually 14 violations of this rule still, a few examples:
[DEFENDER_API_LENGTH_CBOR_PUBLISH](https://github.com/aws/Device-Defender-for-AWS-IoT-embedded-sdk/blob/main/source/defender.c#L162)
Then the rest of those DEFENDER_XXX calls that follow in this switch statement.
Then even with the DNDEBUG it will still flag some of the assert() calls.